cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringProgramEventPublisher.java

```diff
@@ -164,6 +171,7 @@ private PeerProgramUpdates getPeerProgramUpdates() {
         Notification notification = message.decodePayload(
             r -> GSON.fromJson(r, Notification.class));
         if (notification.getNotificationType() != Notification.Type.PROGRAM_STATUS) {
+          lastMessageId = message.getId();
```

Comment:
looks like we can set lastMessageId at the start of the loop instead of in 3 different places

Reply:
fixed